libs/providers/vmware.py

```
@@ -729,6 +729,19 @@ def vm_dict(self, **kwargs: Any) -> dict[str, Any]:
         # Guest OS
         result_vm_info["win_os"] = "win" in vm_config.guestId
 
+        firmware_info: dict[str, Any] = {}
+        boot_firmware = vm_config.firmware.lower()
+        firmware_info["boot_firmware"] = boot_firmware
+        boot_options = vm_config.bootOptions
+        if boot_firmware == "efi" and boot_options is None:
+            raise ValueError(f"bootOptions is None for EFI VM '{_vm.name}' — expected boot configuration")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)